src/psyclone/tests/psyir/backend/sympy_writer_test.py

```diff
@@ -512,3 +512,45 @@ def test_sympy_writer_user_types(fortran_reader, fortran_writer,
     sympy_reader = SymPyReader(sympy_writer)
     new_psyir = sympy_reader.psyir_from_expression(sympy_exp, symbol_table)
     assert fortran_writer(new_psyir) == fortran_expr
+
+
+@pytest.mark.parametrize("expression", ["def", "if", "raise", "del",
+                                        "import", "return", "elif", "in",
+                                        "try", "and", "else", "is", "while",
+                                        "as", "except", "lambda", "with",
+                                        "assert", "finally", "nonlocal",
+                                        "yield", "break", "for", "not",
+                                        "class", "from", "or", "continue",
+                                        "global", "pass"])
+def test_sym_writer_reserved_names(fortran_reader, expression):
+    '''Test that reserved names are properly renamed. In this example,
+    all reserved named will get a ``_1`` appended.
+    '''
+    # A dummy program to easily create the PSyIR for the
+    # expressions we need. We just take the RHS of the assignments
+    source = f'''program test_prog
+                use some_mod
+                integer :: x
+                x = {expression}
+                end program test_prog '''
+    psyir = fortran_reader.psyir_from_source(source)
```

Comment:
This could be shortened if you used `psyir_from_expression` but I won't enforce it as it is already clear what you are doing.

Reply:
I think I tried that. Issue was that `psyir_from_expression` needs a symbol table, and that the symbols need to be declared ... which was a lot more work than just parsing a sample code.

Reply:
I'm happy with that. As I said I think the example is already clear.